addon/components/loading-button.js

```diff
@@ -0,0 +1,41 @@
+import Ember from 'ember';
+
+const {
+  Component,
+  RSVP
+} = Ember;
+
+const LoadingButtonComponent = Component.extend({
+  tagName: 'button',
+
+  click(e) {
+    e.preventDefault();
+
+    // Because `.send` method for sending actions does not return anything, we
+    // pass it an `RSVP.defer` to be resolved in the remote action.
+    let defer = RSVP.defer();
+    defer.promise.then(() => {
+      this.$().text(this.get('originalContent'));
+    }, () => {
```

Comment:
Actually i wanted to ask if you have anything in mind if there are error ... is it to the button to handle it ? 

(And that part is skippable, yes)